plugin.py

```diff
@@ -1,99 +1,88 @@
-        try:
-            if sublime.platform() == 'windows':
-                subprocess.check_call(popenArgs, shell=True)
-            else:
-                subprocess.check_call(popenArgs)
-            onExit()
-        except subprocess.CalledProcessError as error:
-            logAndShowMessage('LSP-json: Error while installing the server.', error)
-        return
-    thread = threading.Thread(target=runInThread, args=(onExit, popenArgs))
-    thread.start()
-    # returns immediately after the thread starts
-    return thread
+    server.setup()
 
 
-def is_node_installed():
-    return shutil.which('node') is not None
+def plugin_unloaded():
+    server.cleanup()
 
 
-def logAndShowMessage(msg, additional_logs=None):
-    print(msg, '\n', additional_logs) if additional_logs else print(msg)
-    sublime.active_window().status_message(msg)
+def is_node_installed():
+    return shutil.which('node') is not None
 
 
 class LspJSONPlugin(LanguageHandler):
+    def __init__(self) -> None:
+        self._default_schemas = []
+
     @property
     def name(self) -> str:
-        return 'lsp-json'
+        return PACKAGE_NAME.lower()
 
     @property
     def config(self) -> ClientConfig:
-        settings = sublime.load_settings("LSP-json.sublime-settings")
-        client_configuration = settings.get('client')
+        # Calling setup() also here as this might run before `plugin_loaded`.
+        # Will be a no-op if already ran.
+        # See https://github.com/sublimelsp/LSP/issues/899
+        server.setup()
+
+        configuration = self.migrate_and_read_configuration()
+
         default_configuration = {
-            "command": [
-                'node',
-                server_path,
-                '--stdio'
-            ],
-            "languages": [
-                {
-                    "languageId": "json",
-                    "scopes": ["source.json"],
-                    "syntaxes": [
-                        "Packages/JavaScript/JSON.sublime-syntax",
-                        "Packages/JSON/JSON.sublime-syntax"
-                    ]
-                }
-            ]
+            'enabled': True,
+            'command': ['node', server.binary_path, '--stdio'],
         }
-        default_configuration.update(client_configuration)
-        default_configuration['settings']['json']['schemas'] = schemas
-        return read_client_config('lsp-json', default_configuration)
+
+        default_configuration.update(configuration)
+
+        if 'json' not in default_configuration['settings']:
+            default_configuration['settings']['json'] = {}
+        if not self._default_schemas:
+            schemas = ['schemas_extra.json', 'schemas.json']
+            for schema in schemas:
+                path = 'Packages/{}/{}'.format(PACKAGE_NAME, schema)
+                self._default_schemas.extend(json.loads(ResourcePath(path).read_text()))
+        default_configuration['settings']['json']['schemas'] = self._default_schemas
```

Comment:
you can use `.setdefault` from the `dict` object here:

```python
default_configuration['settings'].setdefault('json', {})['schemas'] = self._default_schemas
```

this will add `'json': {}` if it doesn't exist yet, and will return the value if `'json'` does exist.